File: 4multiple_inheritance.cpp
#include <bits/stdc++.h>
using namespace std;

class Animal
{
protected:
    string species;

public:
    Animal(string species) : species(species) {}
    void info()
    {
        cout << this->species << " is Animal" << endl;
    }
};

class Pet
{
public:
    string name;
    Pet(string name) : name(name) {}

    void play()
    {
        cout << this->name << " is playing" << endl;
    }
};

class Dog : public Animal, public Pet
{
public:
    string breed;
    Dog(string species, string name, string breed) : Animal(species), Pet(name), breed(breed) {}

    void bark()
    {
        cout << "Dog is barking" << endl;
    }
};

int main()
{

    Dog my_dog("Canine", "Romeo", "Labrador"); 
    my_dog.info();                             
    my_dog.play();                             
    my_dog.bark();
    return 0;
}

File: 4multiple_inheritance.py
# Parent class 1
class Animal:
    def __init__(self, species):
        self.species = species

    def info(self):
        print(f"{self.species} is Animal")

# Parent class 2
class Pet:
    def __init__(self, name):
        self.name = name

    def play(self):
        print(f"{self.name} is playing")

# Child class inheriting from both Animal and Pet
class Dog(Animal, Pet):
    def __init__(self, species, name, breed):
        Animal.__init__(self, species)
        Pet.__init__(self, name)
        self.breed = breed

    def bark(self):
        print("Dog is barking")

# Create an instance of Dog
my_dog = Dog("Canine", "Romeo", "Labrador")
print(my_dog.species)  
print(my_dog.name)     
print(my_dog.breed)    
my_dog.info() 
my_dog.play()         
my_dog.bark()          
